scripts/ci/setup-dev-sso.sh

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env bash
+
+user_help() {
+    echo "Deploy in cluster keycloak and configure registration service to use it."
+    echo "options:"
+    echo "-sn, --sso-ns  Builds and pushes the operator to quay"
+    echo "-h,  --help              To show this help text"
+    echo ""
+    exit 0
+}
+
+read_arguments() {
+    if [[ $# -lt 2 ]]
+    then
+        echo "There are missing parameters"
+        user_help
+    fi
+
+    while test $# -gt 0; do
+           case "$1" in
+                -h|--help)
+                    user_help
+                    ;;
+                -sn|--sso-ns)
+                    shift
+                    DEV_SSO_NS=$1
+                    shift
+                    ;;
+                *)
+                   echo "$1 is not a recognized flag!" >> /dev/stderr
+                   user_help
+                   exit -1
+                   ;;
+          esac
+    done
+}
+
+check_commands()
+{
+    for cmd in "$@"
+    do
+        check_command "$cmd"
+    done
+}
+
+check_command()
+{
+    command -v "$1" > /dev/null && return 0
+
+    printf "please install '%s' before running this script\n" "$1"
+    exit 1
+}
+
+read_arguments $@
+
+if [[ -n "${CI}" ]]; then
+    set -ex
+else
+    set -e
+fi
+
+
+check_commands yq oc base64 openssl
+
+parent_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+cd "$parent_path"
+
+printf "creating ${DEV_SSO_NS} namespace\n"
+cat dev-sso/namespace.yaml | \
+DEV_SSO_NS=${DEV_SSO_NS} envsubst | \
+oc apply -f -
```

Comment:
we can simplify this preferring stdin over `cat`. Something like the following:

```suggestion
DEV_SSO_NS=${DEV_SSO_NS} envsubst < "dev-sso/namespace.yaml" | oc apply -f -
```

Reply:
agree, looks easier to read. Done in https://github.com/codeready-toolchain/toolchain-cicd/pull/113/commits/fd9388eb052b89505519c971d2b167789e827749